internal/broker/helper_test.go

```diff
@@ -15,31 +17,32 @@ import (
 	"github.com/ubuntu/authd-oidc-brokers/internal/broker"
 	"github.com/ubuntu/authd-oidc-brokers/internal/providers/info"
 	"github.com/ubuntu/authd-oidc-brokers/internal/testutils"
+	"github.com/ubuntu/authd-oidc-brokers/internal/token"
 	"golang.org/x/oauth2"
 )
 
 // newBrokerForTests is a helper function to create a new broker for tests with the specified configuration.
 //
-// Note that the IssuerURL is required in the configuration.
+// Note that the issuerURL is required in the configuration.
 func newBrokerForTests(t *testing.T, cfg broker.Config) (b *broker.Broker) {
 	t.Helper()
 
-	require.NotEmpty(t, cfg.IssuerURL, "Setup: issuerURL must not be empty")
+	require.NotEmpty(t, cfg.IssuerURL(), "Setup: issuerURL must not be empty")
 
-	if cfg.CachePath == "" {
-		cfg.CachePath = t.TempDir()
+	if cfg.DataDir == "" {
+		cfg.DataDir = t.TempDir()
 	}
-	if cfg.ClientID == "" {
-		cfg.ClientID = "test-client-id"
+	if cfg.ClientID() == "" {
+		cfg.SetClientID("test-client-id")
 	}
 
 	b, err := broker.New(
 		cfg,
 		broker.WithSkipSignatureCheck(),
 		broker.WithCustomProviderInfo(&testutils.MockProviderInfoer{
 			Groups: []info.Group{
-				{Name: "remote-group", UGID: "12345"},
-				{Name: "linux-local-group", UGID: ""},
+				{Name: "new-broker-for-tests-remote-group", UGID: "12345"},
+				{Name: "linux-new-broker-for-tests-local-group", UGID: ""},
```

Comment:
Why do we need this?

Reply:
From the commit message:

```
Use different group names in different tests
    
To make it easier to figure out where in the code a group name comes
from.
```

I found that useful when I [debugged the failing test](https://github.com/ubuntu/authd-oidc-brokers/pull/134#issuecomment-2348571619).

Reply:
Ack. I thought the `saved-` prefix would be enough, but I see the reason of using a name that's more different to help with "mental overhead".